### PR TITLE
Add preview apache pulsar sink changefeed docs v24.1

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-9664

This PR adds docs for the Apache Pulsar preview sink for v24.1. 

- Changefeed sink description
- Short-form example
- List of unsupported options/features
- Added to the preview section of the CockroachDB Feature Availability page.

Follow the links in the Files changed box below for rendered previews.

### CODE REVIEW

File: src/current/_includes/v24.1/cdc/apache-pulsar-unsupported.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Changefeeds emitting to an Apache Pulsar sink do **not** support:
+
+- [`resolved`]({% link {{ page.version.version }}/create-changefeed.md %}#resolved-option)
```
Comment (kathancox) [inline src/current/_includes/v24.1/cdc/apache-pulsar-unsupported.md:3]: I have included all these options as unsupported, but I was able to start changefeeds with at least the top three into pulsar and receive messages with the respective timestamps (e.g., resolved). Given [Jay's issue](https://github.com/cockroachdb/cockroach/issues/118863), not sure if we want to include these or not. Maybe include until further testing?

**Review by rharding6373 — APPROVED** [review]
Thank you for putting this together! Only have one comment.

**Comment by kathancox** [comment]
> Thank you for putting this together! Only have one comment.

Hey @rharding6373, tftr, I don't see your comment in the PR.

**Review by Amruta-Ranade — APPROVED** [review]
LGTM!

**Pull request merged**